packages/jest/plugins/resolver.ts

```diff
@@ -60,6 +61,13 @@ module.exports = function (path: string, options: ResolveOptions) {
           ...pkg,
           main: pkg.main || pkg.es2015 || pkg.module,
         }),
+        pathFilter: (pkg) => {
+          if (!pkg.exports) {
+            return path;
+          }
+
+          return resolveExports(pkg, path) || path;
+        },
```

Comment:
@barbados-clemens 
I had to add a path filter to the Jest resolver so that it could find the new `exports` field from ESM package.json files.
Could you check that I haven't overlooked anything here or missed any potential edge cases etc?

Reply:
I ran through a handful of tests/checks and everything ran as expected. @Cammisuli said he'll take a look when he gets a chance as well.